src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -66,10 +68,20 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
                     .and()
                     .withProduces(JSON_ONLY)
                     .withResponseType(persistence.clazz),
-                this::deleteApiKeyForApiUser, JsonUtils::toJson
+                this::deleteApiKeyForApiUser, JsonUtils::toJson)
+            // Authenticate user with Auth0
+            .post(path(ID_PATH + AUTHENTICATE_PATH)
+                    .withDescription("Authenticates ApiUser with Auth0.")
+                    .withPathParam().withName(ID_PARAM).withDescription("The user ID.").and()
+                    .withQueryParam().withName(USERNAME_PARAM).withRequired(true)
+                    .withDescription("Auth0 username (usually email address).").and()
+                    .withQueryParam().withName(PASSWORD_PARAM).withRequired(true)
+                    .withDescription("Auth0 password.").and()
+                    .withProduces(JSON_ONLY)
+                    .withResponseType(TokenHolder.class),
```

Comment:
Normally I'd be in favor of exposing a subset of `TokenHolder`... we'll see how that works out.

Reply:
@binh-dam-ibigroup I wasn't sure on this. Originally I was returning just the token, but Landon suggested returning the complete entity (or at least all values returned).